### PR TITLE
White list again in tests

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description
To have again a the tests that have been edited with @dcoeurjo script.
By testing I have just some doubts on the cmake option -E $EXCLUDE_TESTS that looks not always working (not related I think, I will add an issue if it is confirmed).

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [ ] No warning raised in Debug ```cmake``` mode (otherwise, C.I. will fail).
- [ ] All continuous integration tests pass (Github Actions & appveyor)

### CODE REVIEW

**Comment by dcoeurjo** [comment]
👍

**Comment by dcoeurjo** [comment]
why GitHub actions is not starting for this PR? I don't understand

**Comment by kerautret** [comment]
> why GitHub actions is not starting for this PR? I don't understand

Perhaps : on: [push] ? in worksflows file:  buildAndDocumentation.yml ?

**Comment by kerautret** [comment]
> > why GitHub actions is not starting for this PR? I don't understand
> 
> Perhaps : on: [push] ? in worksflows file: buildAndDocumentation.yml ?
By replacing it with it should work I think:
on: [push, pull_request]

File: ChangeLog.md [hunk 1]
```diff
@@ -23,6 +23,8 @@
   - New cmake option (DGTAL_RANDOMIZED_TESTING_THRESHOLD) to set the
     (approximated) % of unit-tests to build and run for randomized
     testing (David Coeurjolly [#1588](https://github.com/DGtal-team/DGtal/pull/1588))
+  - Fix missing white list for the unit-tests in relation to PR [#1591](https://github.com/DGtal-team/DGtal/pull/1591))
```
Comment (dcoeurjo) [inline ChangeLog.md:26]:
```suggestion
  - Fix missing whitelist for the unit-tests in relation to PR [#1591](https://github.com/DGtal-team/DGtal/pull/1591))
```

File: .github/workflows/buildAndDocumentation.yml [hunk 2]
```diff
@@ -25,31 +23,39 @@ jobs:
...
+           sudo apt-get install zsh libqglviewer-dev-qt5 libboost-dev libeigen3-dev ninja-build libhdf5-serial-dev libboost-dev libcairo2-dev libgmp-dev libgraphicsmagick++1-dev libfftw3-dev
 
      - name: Install macOS deps
        if: matrix.os == 'macOS-latest'
        run: brew install boost ninja eigen gmp
 
+     - name: Get white list of tests
+       run: |
+           echo "::set-output name=WHITE_LIST::$(${{runner.workspace}}/DGtal/.github/workflows/create_whitelist.sh)"
+       id: whitelist
```
Comment (dcoeurjo) [inline .github/workflows/buildAndDocumentation.yml:43]: is the id used somewhere?

File: .github/workflows/buildAndDocumentation.yml [hunk 3]
```diff
@@ -25,31 +23,39 @@ jobs:
...
-           sudo apt-get install libqglviewer-dev-qt5 libboost-dev libeigen3-dev ninja-build libhdf5-serial-dev libboost-dev libcairo2-dev libgmp-dev libgraphicsmagick++1-dev libfftw3-dev
+           sudo apt-get install zsh libqglviewer-dev-qt5 libboost-dev libeigen3-dev ninja-build libhdf5-serial-dev libboost-dev libcairo2-dev libgmp-dev libgraphicsmagick++1-dev libfftw3-dev
 
      - name: Install macOS deps
        if: matrix.os == 'macOS-latest'
        run: brew install boost ninja eigen gmp
 
+     - name: Get white list of tests
+       run: |
+           echo "::set-output name=WHITE_LIST::$(${{runner.workspace}}/DGtal/.github/workflows/create_whitelist.sh)"
```
Comment (dcoeurjo) [inline .github/workflows/buildAndDocumentation.yml:42]: is it just an echo or the variable is correctly set ?

File: .github/workflows/buildAndDocumentation.yml [hunk 4]
```diff
@@ -25,31 +23,39 @@ jobs:
...
+
      - name: Create Build Environment
        run: cmake -E make_directory ${{runner.workspace}}/build
 
      - name: Configure CMake (linux)
        if: matrix.os == 'ubuntu-latest'
        shell: bash
        working-directory: ${{runner.workspace}}/build
-       run: cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE $CONFIG_GLOBAL $CONFIG_LINUX -G Ninja
+       run: cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=$BUILD_TYPE $CONFIG_GLOBAL $CONFIG_LINUX -DDGTAL_RANDOMIZED_TESTING_WHITELIST="${{ steps.whitelist.outputs.WHITE_LIST }}" -G Ninja
```
Comment (dcoeurjo) [inline .github/workflows/buildAndDocumentation.yml:52]: I would prefer `WHITELIST` rather than `WHITE_LIST`

**Review by dcoeurjo — CHANGES_REQUESTED** [review]

File: .github/workflows/buildAndDocumentation.yml (hunk 2 above)
Comment (kerautret) [reply .github/workflows/buildAndDocumentation.yml:43]: yes it is used to access this "steps" and to get the stored value named WHITE_LIST

File: .github/workflows/buildAndDocumentation.yml (hunk 3 above)
Comment (kerautret) [reply .github/workflows/buildAndDocumentation.yml:42]: not only, it also store in WHITE_LIST the output of script

File: .github/workflows/buildAndDocumentation.yml (hunk 4 above)
Comment (kerautret) [reply .github/workflows/buildAndDocumentation.yml:52]: Ok I change it, thanks

**Comment by dcoeurjo** [comment]
thanks, let's merge this one

**Pull request merged**